File: src/UI/SDK/src/models/HSShipEstimate.ts
import { ShipEstimateXP } from './ShipEstimateXP';
import { HSShipMethod } from './HSShipMethod';
import { ShipEstimateItem } from './ShipEstimateItem';

export interface HSShipEstimate {
    xp?: ShipEstimateXP
    ShipMethods?: HSShipMethod[]
    ID?: string
    SelectedShipMethodID?: string
    ShipEstimateItems?: ShipEstimateItem[]
}

File: src/Middleware/src/Headstart.Common/Models/Headstart/CheckoutIntegration/HSShipEstimate.cs
using System.Collections.Generic;
using OrderCloud.SDK;

namespace Headstart.Common.Models
{
    public class HSShipEstimate : ShipEstimate<ShipEstimateXP, HSShipMethod>
    {
    }

    public class ShipEstimateXP
    {
        public IList<HSShipMethod> AllShipMethods { get; set; }

        public string SupplierID { get; set; }

        public string ShipFromAddressID { get; set; }
    }
}
